frontend/packages/ceph-storage-plugin/integration-tests/tests/installFlow.scenario.ts

```diff
@@ -0,0 +1,254 @@
+  sourceType: grpc
+---`;
+
+describe('Testing OCS Subscription', () => {
+  beforeAll(async () => {
+    await execSync(`echo '${ocsDeploy}' | oc apply -f -`);
+    // Let the cluster add defns
+    await browser.sleep(45000);
+  }, 100000);
+
+  it('Subscribe to Operator', async () => {
+    await browser.wait(until.and(crudView.untilNoLoadersPresent));
+    await goToOperatorHub();
+    await browser.wait(until.and(crudView.untilNoLoadersPresent));
+    const ocs = await searchInOperatorHub('Openshift Container Storage');
+    ocs.click();
+    let uninstall = null;
+    try {
+      uninstall = !!(await uninstallOperator.isDisplayed());
+    } catch {
+      uninstall = false;
+    }
+    if (uninstall) {
+      // eslint-disable-next-line
+      console.log('Already subscribed to the operator');
+      await browser.wait(until.visibilityOf(closeOperatorInstall));
+      await closeOperatorInstall.click();
+      await operatorsPage();
+    } else {
+      await browser.wait(until.visibilityOf(installOperator));
+      await installOperator.click();
+      browser.refresh();
+      await installer.installOperator();
+    }
+    await browser.wait(until.visibilityOf(ocsOperator));
+    expect(ocsOperator.isDisplayed()).toBe(true);
+    await browser.wait(until.visibilityOf(ocsOperatorStatus));
+    let text = await ocsOperatorStatus.getText();
+    // If not installed yet wait for 15 seconds
+    if (!text.includes('Succeeded')) await browser.sleep(10000);
+    text = await ocsOperatorStatus.getText();
+    expect(text.includes('Succeeded') || text.includes('date')).toBe(true);
+  });
+
+  it(
+    'Create storage cluster',
+    async () => {
+      await operatorsPage();
+      await browser.wait(until.visibilityOf(ocsOperator));
+      await ocsOperator.click();
+      await browser.wait(until.visibilityOf(storageClusterLink));
+      await storageClusterLink.click();
+      await browser.wait(until.and(crudView.untilNoLoadersPresent));
+      // Node list fluctautes
+      await browser.sleep(3000);
+      await selectWorkerRows();
+      // Fluctating
+      await browser.sleep(1000);
+      await browser.wait(until.elementToBeClickable(btnCreate));
+      await btnCreate.click();
+      // Let it move to the other page
+      await browser.sleep(1000);
+      await browser.wait(until.visibilityOf(status));
+      await browser.wait(until.visibilityOf(crudView.resourceTitle));
+      const text = await crudView.resourceTitle.getText();
+      expect(text).toEqual('ocs-storagecluster');
+      // Wait for cluster to come to ready state
+      checkIfClusterIsReady();
+    },
+    15 * 60 * 1000,
+  );
+});
+
+describe('Tests if all the PODS are up', () => {
+  let pods = null;
+
+  beforeAll(async () => {
+    pods = await JSON.parse(
+      execSync('kubectl get po -n openshift-storage -o json').toString('utf-8'),
+    );
+  });
+
+  it('Tests if ocs operator pods is running', async () => {
+    const pattern = 'ocs-operator-';
```

Comment:
Wouldn't it be nicer to have a function for checking if a specific pod (or a specific number of pods that match the pattern) are running and then using that function in each of the tests?

Reply:
This would give us a chance for tests to localize what pod is no working. Are you suggesting we pus the expect statements inside some function?